### PR TITLE
Add functionality from tf_slim to replace tf.contrib functionality

### PR DESCRIPTION
**Pull request by abhishyantkhare**
TLDR: Google sucks, this was harder than it needed to be, but the TF2
issues should be solved now.

The issue here was that usage of tf.contrib cannot be replaced
automatically by the upgrade script. It seems that several libraries in
tf.contrib have not been automatically ported. Google really bungled the
upgrade process from TF1 to TF2. They have not managed backwards
compatability well with little to no thought to existing users. This
was not an easy task to solve, and the only entity to blame here is
Google. You can track the upgrade process of TF1 contrib functionality
here:
https://docs.google.com/spreadsheets/d/1FLFJLzg7WNP6JHODX5q8BDgptKafq_slHpnHVbJIteQ/edit#gid=0
(Notice how empty the sheet is)

Fortunately, the functionality required by astronet was ported to a
separate library Google created called tf-slim. There is no
documentation that they did that, and it is instead left to the user to
discover that. Why not include that information in the TF2 upgrade page?
Who knows why Google seems to hate their users?

However, it's not just as easy pip installing tf-slim, because while
the GitHub repository has updated code, the actual pip package is not
updated and still contains references to tf.contrib. Why not update the
pip package if its such a major release? No good reason other than pure
mismanagement of this upgrade process. I don't know if anyone at Google
really even thought through this well.

Anyways, I managed to cherry pick the exact functionality astronet
needed from tf-slim and put it into astronet itself and modified the
imports. Everything should work correctly now.

### CODE REVIEW

**Comment by drpiccioni** [comment]
Merging Abhi's amazing work overcoming the missteps of Google.
